OpenMP detection broken when building with -coverage

Detection of OpenMP support when building with the `-coverage` option (see #374) is broken again because the `CFLAGS` environment variable is propagated to the compiler invocation but not the linker invocation. The environment variable `CFLAGS=-coverage` normally tells the build process to add the `-coverage` option to both compiler AND linker commands for C binaries, but as shown below it is only passed to the compiler. As a result, the executable fails to run.

```
gcc -pthread -DNDEBUG -g -fwrapv -O3 -Wall -coverage -fPIC -c test_openmp.c -o objects/test_openmp.o -fopenmp
gcc -pthread objects/test_openmp.o -o test_openmp -fopenmp
objects/test_openmp.o: In function `_GLOBAL__sub_I_65535_0_test_openmp.c':
test_openmp.c:(.text.startup+0x38): undefined reference to `__gcov_init'
objects/test_openmp.o:(.data.rel+0x20): undefined reference to `__gcov_merge_add'
collect2: error: ld returned 1 exit status
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -13,7 +13,7 @@ astropy-helpers Changelog
 3.1.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Fixed broken OpenMP detection when building with ``-coverage``. [#434]
 
 
 3.1 (2018-12-04)

--- a/astropy_helpers/openmp_helpers.py
+++ b/astropy_helpers/openmp_helpers.py
@@ -176,6 +176,11 @@ def check_openmp_support(openmp_flags=None):
     compile_flags = openmp_flags.get('compiler_flags')
     link_flags = openmp_flags.get('linker_flags')
 
+    # Pass -coverage flag to linker.
+    # https://github.com/astropy/astropy-helpers/pull/374
+    if '-coverage' in compile_flags and '-coverage' not in link_flags:
+        link_flags.append('-coverage')
+
     tmp_dir = tempfile.mkdtemp()
     start_dir = os.path.abspath('.')
 

